src/index.ts

```diff
@@ -249,42 +232,40 @@ export function ruleFactory<
           stepRow,
           stepCount,
         });
-        break;
-      } else if ('try' in rule) {
+        // eslint-disable-next-line @typescript-eslint/no-throw-literal
+        throw BREAK;
+      } else if ('try' in rule && 'catch' in rule) {
         try {
-          const tryResult = evaluateRule({
-            stepRow,
-            input,
-            rule: rule.try,
-            ignoreMissingKeys: true,
-          });
-          results.lastValue = tryResult;
           logTrace({
             operation: 'try',
             rule: rule.try,
-            result: serialize(tryResult),
             currentState: serialize(input),
             stepRow,
             stepCount,
           });
+          handleRule(rule.try);
         } catch (e) {
-          const catchResult = evaluateRule({
-            stepRow,
-            input,
-            rule: rule.catch,
-            ignoreMissingKeys: true,
-          });
-          results.lastValue = catchResult;
           logTrace({
             operation: 'catch',
             rule: rule.catch,
-            result: serialize(catchResult),
             currentState: serialize(input),
             stepRow,
             stepCount,
           });
+          handleRule(rule.catch);
         }
       }
+    };
+
+    rules = arrayify(rules);
+
+    for (const rule of rules) {
+      try {
+        handleRule(rule);
```

Comment:
Could we rely on a `return` result, rather than try/catch for both errors & early exits?

```ts
for (const rule of rules) {
  try {
    const result = handleRule(rule);
    if (result === 'break') break;
  } catch (error) {
    // add logTrace
    throw error;
  }
}